modules/storages/app/common/storages/peripherals/storage_url_helper.rb

```diff
@@ -38,78 +38,5 @@ def storage_url_open_file(file_link, open_location: false)
     def storage_url_open(storage)
       "#{storage.host}/index.php/apps/files"
     end
-
-    # rubocop:disable Metrics/AbcSize
-    def make_download_url(file_link:, user:)
-      storage = file_link.storage
-      oauth_client = storage.oauth_client
-
-      client_token = get_oauth_client_token(user:, oauth_client:)
-      return client_token if client_token.failure?
-
-      direct_download_response = make_direct_download storage:,
-                                                      access_token: client_token.result.access_token,
-                                                      file_id: file_link.origin_id
-
-      return direct_download_response if direct_download_response.failure?
-
-      download_token = direct_download_token(body: direct_download_response.result)
-      return download_token if download_token.failure?
-
-      url = "#{storage.host}/index.php/apps/integration_openproject/direct/#{download_token.result}/#{file_link.origin_name}"
-      ServiceResult.success(result: url)
-    end
-
-    # rubocop:enable Metrics/AbcSize
-
-    private
-
-    def get_oauth_client_token(user:, oauth_client:)
-      client_token = ::OAuthClients::ConnectionManager
-                       .new(user:, oauth_client:)
-                       .get_access_token
-
-      client_token.success? ? client_token : ServiceResult.failure(result: I18n.t('http.request.missing_authorization'))
-    end
-
-    def make_direct_download(storage:, access_token:, file_id:)
-      response = Storages::Peripherals::StorageRequests
-                   .new(storage:)
-                   .download_command
-                   .call(access_token:, file_id:)
-
-      return response if response.failure?
-
-      # The nextcloud API returns a successful response with empty body if the authorization is missing or expired
```

Comment:
@Kharonus Not sure if I need to address this anymore. What do you think.

Reply:
uff, we need to check this.

Reply:
Sadly, yes. If we post to the direct download endpoint to get our token, but with any invalid oauth token (e.g. `Bearer 123`), we get this kind of response:

```
https://nextcloud.local/ocs/v2.php/apps/dav/api/v1/direct

HTTP/1.1 200 OK
Cache-Control: no-store, no-cache, must-revalidate
Content-Length: 0
Content-Security-Policy: default-src 'self'; script-src 'self' 'nonce-QWplKzJnSWQ1SHJCM2ZTYXN5akhESzM3Sm9uRXZaV3o2WXlRRkx1ZTZiOD06TkFIM3RVMDJxZzZ5anFLeDVrV1BYLzZ4WDhldCsrLyswT2plWmZqa2tPZz0='; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *; object-src 'none'; base-uri 'self';
Content-Type: text/html; charset=UTF-8
Date: Tue, 15 Nov 2022 15:31:46 GMT
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Pragma: no-cache
Referrer-Policy: no-referrer
Server: Apache/2.4.54 (Debian)
Set-Cookie: oc9cwt1ct0fo=13c4f47ddf31864c6f224ea0c0dd1d7c; path=/; secure; HttpOnly; SameSite=Lax
Set-Cookie: oc_sessionPassphrase=hNhM3VQjkzQhnxEQyUDqyA0h3OAxa0xtoq9%2BJ%2BmaWuglrvlbAKKk9y9kDrMwrV6HJCl%2B2zSTFVZwTr2%2FNwUru%2BK%2FZpCPLBLFCVdRtfvBk7hVFomvTJNUbjpfn7%2BZl2eb; path=/; secure; HttpOnly; SameSite=Lax
Set-Cookie: oc9cwt1ct0fo=718a15563ab1e4722ba4a604060fe28e; path=/; secure; HttpOnly; SameSite=Lax
Set-Cookie: __Host-nc_sameSiteCookielax=true; path=/; httponly;secure; expires=Fri, 31-Dec-2100 23:59:59 GMT; SameSite=lax
Set-Cookie: __Host-nc_sameSiteCookiestrict=true; path=/; httponly;secure; expires=Fri, 31-Dec-2100 23:59:59 GMT; SameSite=strict
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Permitted-Cross-Domain-Policies: none
X-Powered-By: PHP/8.0.21
X-Robots-Tag: none
X-Xss-Protection: 1; mode=block

<Response body is empty>

Response code: 200 (OK); Time: 44ms (44 ms); Content length: 0 bytes (0 B)
```